src/tasks/minions/pickpocketActivity.ts

```diff
@@ -0,0 +1,102 @@
+import { percentChance } from 'e';
+import { Task } from 'klasa';
+import { Bank } from 'oldschooljs';
+
+import { Events, Time } from '../../lib/constants';
+import { Pickpockable, Pickpocketables } from '../../lib/skilling/skills/thieving/stealables';
+import { SkillsEnum } from '../../lib/skilling/types';
+import { PickpocketActivityTaskOptions } from '../../lib/types/minions';
+import createReadableItemListFromBank from '../../lib/util/createReadableItemListFromTuple';
+import { handleTripFinish } from '../../lib/util/handleTripFinish';
+
+export function calcLootXPPickpocketing(
+	currentLevel: number,
+	npc: Pickpockable,
+	quantity: number
+): [number, number, number, number] {
+	let xpReceived = 0;
+
+	let successful = 0;
+	let damageTaken = 0;
+	// Pickpocketing takes 2 ticks
+	const timeToPickpocket = (npc.customTickRate ?? 2) * 0.6;
+	// For future Ardougne Diary and Thieving cape
+	const diary = 1;
+	const thievCape = 1;
+
+	const chanceOfSuccess = (npc.slope * currentLevel + npc.intercept) * diary * thievCape;
+
+	for (let i = 0; i < quantity; i++) {
+		if (!percentChance(chanceOfSuccess)) {
+			// The minion has just been stunned, and cant pickpocket for a few ticks, therefore
+			// they also miss out on the next few pickpockets depending on stun time. And take damage
+			damageTaken += npc.stunDamage;
+			quantity -= Math.round(npc.stunTime / timeToPickpocket);
```

Comment:
actually youd need to subtract an additional 1 here in order to use quantity as a counter for successful pickpockets